src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,40 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+    constructor({ popupSelector, handleFormSubmit }) {
+        super({ popupSelector });
+        this._popupForm = this._popup.querySelector(".modal__delete-form");
+        this._submitButton = this._popupForm.querySelector(".modal__delete-button")
+        this._defaultButtonText = this._submitButton
+        ? this._submitButton.textContent
+        : "";
+        this._handleFormSubmit = handleFormSubmit;
+        this._deleteCardForm = document.forms["delete-card-form"];
```

Comment:
`        this._deleteCardForm = document.forms["delete-card-form"];
`

You don't need it because you have ` this._popupForm` in line 6